BizHawk.Client.Common/movie/tasproj/TasStateManager.cs

```diff
@@ -26,6 +26,27 @@ private void CallInvalidateCallback(int index)
 			InvalidateCallback?.Invoke(index);
 		}
 
+		public Action<int> LoadCallback { get; set; }
+
+		private void CallLoadCallback(int index)
+		{
+			LoadCallback?.Invoke(index);
+		}
+
+		public Action<int> SaveCallback { get; set; }
+
+		private void CallSaveCallback(int index)
+		{
+			SaveCallback?.Invoke(index);
+		}
+
+		public Action<int> RemoveCallback { get; set; }
+
+		private void CallRemoveCallback(int index)
+		{
+			RemoveCallback?.Invoke(index);
+		}

```

Comment:
All this logic should be in https://github.com/TASVideos/BizHawk/blob/master/BizHawk.Client.EmuHawk/tools/TAStudio/BookmarksBranchesBox.cs

StateManager is only managing savestates related to branches, bot branch operations themselves, which are a few levels above, and separated from Common.

Reply:
I'm not sure how to do that. The functions in BookmarksBranchesBox don't have branch index

Reply:
You can do `Movie.Branches.IndexOf(branch);` there.